modules/subscriptions.php

```diff
@@ -222,6 +225,20 @@ function maybe_send_subscription_email( $new_status, $old_status, $post ) {
 				update_post_meta( $post->ID, '_jetpack_dont_email_post_to_subs', $_POST['_jetpack_dont_email_post_to_subs'] );
 			}
 		}
+
+		if ( ! get_post_meta( $post->ID, '_jetpack_dont_email_post_to_subs' ) ) {
+			self::$send_email = $post->ID;
+		}
+
+	}
+
+	function maybe_trigger_send_email_action( $post_ID ) {
```

Comment:
It seems that this function is missing a lot of logic from the legacy `maybe_send_subscription_email` function, for example:

- excluding posts whose categories are in the `jetpack_subscriptions_exclude_these_categories` filter
- excluding posts whose categories are ONLY in the `jetpack_subscriptions_exclude_all_categories_except` filter, if it's not empty
- checking for `$_POST['_jetpack_dont_email_post_to_subs']` to disable sending to subscribers for this particular save
- checking if the `_jetpack_dont_email_post_to_subs` is already set (e.g. because it was saved with the "don't email" checkbox checked previously